plugins/linear-genome-view/src/BaseLinearDisplay/models/BaseLinearDisplayModel.tsx

```diff
@@ -263,16 +405,22 @@ export const BaseLinearDisplay = types
      */
     regionCannotBeRendered(_region: Region) {
       const view = getContainingView(self) as LinearGenomeViewModel
-      if (view && view.bpPerPx > self.maxViewBpPerPx) {
+      if (
+        view &&
+        self.globalStats.featureDensity !== undefined &&
+        self.globalStats.featureDensity > self.maxFeatureScreenDensity
+      ) {
         return (
           <>
             <Typography component="span" variant="body2">
-              Zoom in to see features or{' '}
+              Increase max feature screen density or{' '}
```

Comment:
I think we may want to keep the same text, "Zoom in to see features", since this is the easiest thing for the user to do

Reply:
gonna mark this as discuss in meeting, for implementation @rbuels wanted a cache so that stats estimation would only run once, and with the cache that means that zooming in doesn't recalcuate the feature density so there would be no change

Reply:
could discuss, but one way to think of it is that if you are zoomed in, less features will be on the screen, while when you are zoomed out, there will be more features on the screen. the feature density tells us an estimate of how many features there for every base pair on the screen, so it helps us determine, essentially, how many features will be on the screen at a given zoom level (e.g. if you are viewing a region that is 100,000bp, then the featureDensity*100,000 will equal the estimated number of features in that region. if that number is large, we can say, we don't want to render that many features, please zoom in). here is some code jbrowse 1 uses to determine whether to render the features. scale is equal to 1/view.bpPerPx https://github.com/GMOD/jbrowse/blob/dev/src/JBrowse/View/Track/CanvasFeatures.js#L260